### PR TITLE
Users management and Observer Feature

### PR DESCRIPTION
**Pull request by ddomingues**
This will enable the admin:

* Management the users of the system: editing, showing and destroying;
* Sign in as an observer to see the progress or a possible problem of the client certification process

### CODE REVIEW

File: app/views/admin/users/show.html.slim [hunk 1]
```diff
@@ -0,0 +1,10 @@
+= main_panel_heading title: crud_title, instance: @user, actions: [:delete, :edit]
+= main_panel_alert type: :danger, message: @user.errors.full_messages.join(", ") if @user.errors.any?
+= describe_container(horizontal: true, class: "margin-top-20 margin-bottom-10") do
+  = describe resource: @user, field: :name
+  = describe resource: @user, field: :email
+  = describe resource: @user, field: :admin, value: @user.admin.to_s
+  = describe resource: @user, field: :last_sign_in_at, value: l(@user.last_sign_in_at, format: :long)
```
Comment (roalcantara) [inline app/views/admin/users/show.html.slim]: Hey @ddomingues, what would happen if we try to show a user with a nil  `last_sign_in_at`? Could encapsulate this check on the `describe` helper be an option?

File: spec/helpers/tablefier_helper_spec.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@
       let(:clazz) { User }
       let(:collection) { create_list :segment, 2 }
       let(:collumns) { [:name, :enabled] }
-      let(:actions) { [:show, :edit, :delete] }
+      let(:actions) { [:show, :edit, :delete, Proc.new { |s| link_to "Show as Proc", admin_segment_path(s) }] }
```
Comment (roalcantara) [inline spec/helpers/tablefier_helper_spec.rb]: 🤘🏼

File: spec/helpers/application_helper_spec.rb [hunk 3]
```diff
@@ -252,4 +252,26 @@ def has_not_updated_by?(track)
...
     end
   end
+
+  describe "#signed_as_an_observer?" do
+    context "when there is an observer" do
+      before do
+        allow(controller).to receive(:user_session).and_return(ApplicationController::OBSERVER_ID.to_s => 1)
+      end
+
+      it "returns true" do
```
Comment (roalcantara) [inline spec/helpers/application_helper_spec.rb]: When the message is really obvious and simple, like this, is it a bad pattern to get rid of the description? For instance, if I have:
```
context "when there is an observer" do
      before do
        allow(controller).to receive(:user_session).and_return(ApplicationController::OBSERVER_ID.to_s => 1)
      end

      it do
        expect(helper.signed_as_an_observer?).to be_truthy
      end
    end
```
My default output is:
```
#signed_as_an_observer?
    when there is an observer
      should be truthy
    when there isn't an observer
      should be falsey
```
Is this a bad approach?

File: app/controllers/admin/users_controller.rb [hunk 4]
```diff
@@ -1,4 +1,4 @@
-class Admin::UsersController < ApplicationController
+class Admin::UsersController < Admin::BaseController
```
Comment (roalcantara) [inline app/controllers/admin/users_controller.rb]: Just to understand: why did `Admin::UsersController` extended `Admin::BaseController` only from now on?

File: app/views/admin/users/show.html.slim (hunk 1 above)
Comment (ddomingues) [inline app/views/admin/users/show.html.slim]: Humm.... I dunno... Seeing it now. 

`Could encapsulate this check on the describe helper be an option?`
Maybe

Comment (ddomingues) [inline app/views/admin/users/show.html.slim]: You're sure. I received an error.

I'll just add a conditional here, besides that we can improve the `describe` method to identify the type of the attribute and apply the correct format to each one.

How about?

**Pull request merged**